packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -495,10 +523,13 @@ export class TerminalFrontendContribution implements TerminalService, CommandCon
         if (!widget.isAttached) {
             this.shell.addWidget(widget, op.widgetOptions);
         }
-        if (op.mode === 'activate') {
-            this.shell.activateWidget(widget.id);
-        } else if (op.mode === 'reveal') {
-            this.shell.revealWidget(widget.id);
+        if (op.mode === 'activate' || op.mode === 'reveal') {
+            if (op.mode === 'activate') {
+                this.shell.activateWidget(widget.id);
+            } else {
+                this.shell.revealWidget(widget.id);
+            }
+            this.setLastUsedTerminal(widget);
```

Comment:
I am not sure that's a correct way to set it, someone could call `shell.revealWidget` for a terminal from other code path and we will miss it. Maybe we should rely on `onDidChangeVisibility` event `isHidden` flag instead?

Reply:
Let me try. My first guess is that ‘revealWidget’ does not mutate the visibility.

Reply:
It’s the purpose of revealwidget.

Reply:
I implemented what you proposed. You are right - it works. Thank you for the suggestion !